src/hotspot/share/opto/block.cpp

```diff
@@ -1221,6 +1237,9 @@ void PhaseCFG::verify() const {
       if (j >= 1 && n->is_Mach() && n->as_Mach()->ideal_Opcode() == Op_CreateEx) {
         assert(j == 1 || block->get_node(j-1)->is_Phi(), "CreateEx must be first instruction in block");
       }
+      if (n->is_memory_writer()) {
+        verify_memory_writer_placement(block, n);
```

Comment:
Other calls to this method are guarded by `#ifdef ASSERT`. Would it make sense to move the `n->is_memory_writer()` into the method, always call it and define it as `NOT_DEBUG_RETURN`?

Reply:
Done, thanks!